core/drivers/gic.c

```diff
@@ -530,13 +535,27 @@ static void gic_init_base_addr(paddr_t gicc_base_pa, paddr_t gicd_base_pa,
 		gd->chip.dt_get_irq = gic_dt_get_irq;
 }
 
+static void gic_init_pre_prop(void)
+{
+	struct gic_data *gd = &gic_data;
+
+	gd->pre_prio = calloc(gd->max_it + 1, sizeof(*gd->pre_prio));
```

Comment:
This will only be needed in very specific configurations, but like this, it will impact all configurations. How about keeping this information outside the GIC driver and letting `interrupt_release()` take the needed parameters to restore the state?

Reply:
Add the gic_spi_get_prop API to get the interrupt properties, and the interrupt_release need to specify the interrupt attributes as parameters.